pallets/stateful-storage/src/lib.rs

```diff
@@ -273,8 +273,14 @@ pub mod pallet {
 		) -> DispatchResult {
 			let key = ensure_signed(origin)?;
 			let is_pruning = actions.iter().any(|a| matches!(a, ItemAction::Delete { .. }));
-			Self::check_schema_for_write(schema_id, PayloadLocation::Itemized, false, is_pruning)?;
-			Self::check_msa_and_grants(key, state_owner_msa_id, schema_id)?;
+			let caller_msa_id = Self::check_msa_and_grants(key, state_owner_msa_id, schema_id)?;
+			let is_signed = caller_msa_id == state_owner_msa_id;
```

Comment:
(reminder before switching out of draft) We should note in the docs of the `_with_signature` and perhaps also of these how this works to cross reference it.

Reply:
Updated docs.